src/qibo/tests/test_quantum_info_random.py

```diff
@@ -0,0 +1,430 @@
+    with pytest.raises(TypeError):
+        q, depth = 1, "1"
+        random_pauli(q, depth)
+    with pytest.raises(ValueError):
+        q, depth = 1, 0
+        random_pauli(q, depth)
+    with pytest.raises(TypeError):
+        q, depth, max_qubits = 1, 1, "1"
+        random_pauli(q, depth, max_qubits=max_qubits)
+    with pytest.raises(ValueError):
+        q, depth, max_qubits = 1, 1, 0
+        random_pauli(q, depth, max_qubits=max_qubits)
+    with pytest.raises(ValueError):
+        q, depth, max_qubits = 4, 1, 3
+        random_pauli(q, depth, max_qubits=max_qubits)
+    with pytest.raises(ValueError):
+        q = [0, 1, 3]
+        depth = 1
+        max_qubits = 2
+        random_pauli(q, depth, max_qubits=max_qubits)
+    with pytest.raises(TypeError):
+        q, depth = 1, 1
+        random_pauli(q, depth, return_circuit="True")
+    with pytest.raises(TypeError):
+        q, depth = 2, 1
+        subset = np.array([0, 1])
+        random_pauli(q, depth, subset=subset)
+    with pytest.raises(TypeError):
+        q, depth = 2, 1
+        subset = ["I", 0]
+        random_pauli(q, depth, subset=subset)
+    with pytest.raises(TypeError):
+        q, depth = 1, 1
+        random_pauli(q, depth, seed=0.1)
+
+
+def test_pauli_single(backend):
+    result = np.array([[1.0 + 0.0j, 0.0 + 0.0j], [0.0 + 0.0j, -1.0 + 0.0j]])
+    result = backend.cast(result, dtype=result.dtype)
+    matrix = random_pauli(0, 1, 1, seed=10).unitary()
+    matrix = backend.cast(matrix, dtype=matrix.dtype)
+    backend.assert_allclose(
+        backend.calculate_norm(matrix - result) < PRECISION_TOL, True
+    )
+
+
+@pytest.mark.parametrize("qubits", [2, [0, 1], np.array([0, 1])])
+@pytest.mark.parametrize("depth", [2])
+@pytest.mark.parametrize("max_qubits", [None])
+@pytest.mark.parametrize("subset", [None, ["I", "X"]])
+@pytest.mark.parametrize("return_circuit", [True, False])
+@pytest.mark.parametrize("seed", [10])
+def test_random_pauli(backend, qubits, depth, max_qubits, subset, return_circuit, seed):
+    result_complete_set = np.array(
+        [
+            [0.0 + 0.0j, 1.0 + 0.0j, 0.0 + 0.0j, 0.0 + 0.0j],
+            [1.0 + 0.0j, 0.0 + 0.0j, 0.0 + 0.0j, 0.0 + 0.0j],
+            [0.0 + 0.0j, 0.0 + 0.0j, 0.0 + 0.0j, 1.0 + 0.0j],
+            [0.0 + 0.0j, 0.0 + 0.0j, 1.0 + 0.0j, 0.0 + 0.0j],
+        ]
+    )
+    result_complete_set = backend.cast(
+        result_complete_set, dtype=result_complete_set.dtype
+    )
+    result_subset = np.eye(4)  # np.array(
+    #     [
+    #         [1.0 + 0.0j, 0.0 + 0.0j, 0.0 + 0.0j, 0.0 + 0.0j],
+    #         [0.0 + 0.0j, 1.0 + 0.0j, 0.0 + 0.0j, 0.0 + 0.0j],
+    #         [0.0 + 0.0j, 0.0 + 0.0j, 0.0 + 0.0j, 1.0 + 0.0j],
+    #         [0.0 + 0.0j, 0.0 + 0.0j, 1.0 + 0.0j, 0.0 + 0.0j],
+    #     ]
+    # )
+    result_subset = backend.cast(result_subset, dtype=result_subset.dtype)
+
+    # qubits = backend.cast(qubits, dtype=qubits.dtype) if isinstance(qubits, np.ndarray) else qubits
```

Comment:
Same here, I don't think this is needed.

Reply:
removed it, thanks